src/naive/internals.rs

```diff
@@ -317,22 +319,22 @@ impl Of {
     }
 
     #[inline]
-    pub(super) fn weekday(&self) -> Weekday {
+    pub(super) const fn weekday(&self) -> Weekday {
         let Of(of) = *self;
-        Weekday::from_u32(((of >> 4) + (of & 0b111)) % 7).unwrap()
+        Weekday::from_u32_mod7((of >> 4) + (of & 0b111))
```

Comment:
Can't use `Weekday::from_u32` because that is part of the `num_traits::FromPrimitive`, and trait methods can't yet be const.